proxy/src/main.rs

```diff
@@ -89,13 +95,17 @@ fn main() -> anyhow::Result<()> {
         (None, None) => None,
         _ => bail!("either both or neither ssl-key and ssl-cert must be specified"),
     };
-
+    let tcp_keepalive = match arg_matches.value_of("keepalive") {
+        Some(ms) => Some(ms.parse()?),
+        None => None,
+    };
```

Comment:
Perhaps you could do `arg_matches.value_of("keepalive").map(str::parse).transpose()`.

Reply:
Yes, it is more compact and readable.